spec/ruby/core/string/chilled_string_spec.rb

```diff
@@ -0,0 +1,46 @@
+require_relative '../../spec_helper'
+
+describe "chilled String" do
+  guard -> { ruby_version_is "3.4" and !"test".equal?("test") } do
+    describe "#frozen?" do
+      it "returns true" do
+        "chilled".frozen?.should == true
+      end
+    end
+
+    describe "#-@" do
+      it "returns a different instance" do
+        input = "chilled"
+        interned = (-input)
+        interned.frozen?.should == true
+        interned.object_id.should_not == input.object_id
+      end
+    end
+
+    describe "#+@" do
+      it "returns a different instance" do
+        input = "chilled"
+        duped = (+input)
+        duped.frozen?.should == false
+        duped.object_id.should_not == input.object_id
+      end
+    end
+
+    describe "mutation" do
+      it "emits a warning" do
```

Comment:
I think it would be good if there was also a spec which asserted that the mutation actually sticks (i.e. something like:)

```
input = "chilled"
input << "mutated"
input.should eql("chilledmutated")
```

I opened the spec first when reviewing this to try and understand precicely what the behaviour was supposed to be and it took me a few goes to figure out that chilled strings are `frozen?` but still actually mutable (with a warning).

Reply:
Added.